src/Servers/Kestrel/Core/src/KestrelConfigurationLoader.cs

```diff
@@ -29,17 +28,28 @@ internal KestrelConfigurationLoader(KestrelServerOptions options, IConfiguration
         {
             Options = options ?? throw new ArgumentNullException(nameof(options));
             Configuration = configuration ?? throw new ArgumentNullException(nameof(configuration));
-            ConfigurationReader = new ConfigurationReader(Configuration);
         }
 
         public KestrelServerOptions Options { get; }
-        public IConfiguration Configuration { get; }
-        internal ConfigurationReader ConfigurationReader { get; }
+        public IConfiguration Configuration { get; internal set; }
+
+        /// <summary>
+        /// If <see langword="true"/>, Kestrel will reload endpoints defined in the "Endpoints" section of the <see cref="Configuration"/>, if they are changed.
+        /// Kestrel will bind and unbind to endpoints as necessary.
+        /// </summary>
+        /// <remarks>Code-defined endpoints defined it code will not be reloaded even if the default certificate or protocol used by endpoint changed.</remarks>
```

Comment:
```suggestion
        /// <remarks>If set the <see langword="true" />, Kestrel will dynamically update endpoint bindings when configuration changes. This will only reload endpoints defined in the "Endpoints" section of your Kestrel configuration. Endpoints defined in code will not be reloaded.</remarks>
```